internal/database/gitserver_repos.go

```diff
@@ -117,6 +124,78 @@ func (s *gitserverRepoStore) Update(ctx context.Context, repos ...*types.Gitserv
 	return errors.Wrap(err, "updating GitserverRepo")
 }
 
+const updatePoolRepoIDQueryFmtStr = `
+-- find the repo that matches the poolRepoName
+WITH pool AS (
+	SELECT
+		id
+	FROM
+		repo
+	WHERE
+		name = %s::citext
+),
+-- find the repo that matches repoName
+repo AS (
+	SELECT
+		id
+	FROM
+		repo
+	WHERE
+		name = %s::citext
+)
+-- now update the pool_repo_id of repo with the repo.id of the pool
+UPDATE
+	gitserver_repos
+SET
+	pool_repo_id = (SELECT id FROM pool)
+WHERE
+	repo_id = (SELECT id FROM repo)
+`
+
+// UpdatePoolRepoID updates the repo matching `repoName` with the repo ID of the repo matching
+// `poolRepoName`.
+func (s *gitserverRepoStore) UpdatePoolRepoID(ctx context.Context, poolRepoName, repoName api.RepoName) error {
+	err := s.Exec(ctx, sqlf.Sprintf(updatePoolRepoIDQueryFmtStr, poolRepoName, repoName))
+	return errors.Wrap(err, "UpdatePoolRepoID: failed to add pool_repo_id to gitserver_repos row")
+}
+
+const getPoolRepoQueryFmtStr = `
+WITH gs AS (
+	SELECT
+		gitserver_repos.pool_repo_id
+	FROM
+		gitserver_repos
+		JOIN repo AS r ON repo_id = r.id
```

Comment:
`JOIN` is on the same line as `FROM`, `WHERE` and others so it's clearly visible that a `JOIN` is going on 😛 

```suggestion
	JOIN repo AS r ON repo_id = r.id
```

Reply:
Because no machine can reach artisanal, hand-crafted SQL formatting yet

Reply:
Truer words were never said. :P